Neos.Media.Browser/Classes/Domain/ImageMapper.php

```diff
@@ -0,0 +1,100 @@
+<?php
+namespace Neos\Media\Browser\Domain;
+
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Flow\ResourceManagement\ResourceManager;
+use Neos\Media\Domain\Model\Adjustment\CropImageAdjustment;
+use Neos\Media\Domain\Model\ImageInterface;
+use Neos\Media\Domain\Model\ImageVariant;
+
+/**
+ *
+ */
+class ImageMapper
+{
+    /**
+     * @var array
+     */
+    private $mappingResult = [];
+
+    /**
+     * @var ResourceManager
+     */
+    protected $resourceManager;
+
+    /**
+     * @var PersistenceManagerInterface
+     */
+    protected $persistenceManager;
+
+    /**
+     * ImageMapper constructor.
+     *
+     * @param ImageInterface $image
+     * @param ResourceManager $resourceManager
+     * @param PersistenceManagerInterface $persistenceManager
+     */
+    public function __construct(ImageInterface $image, ResourceManager $resourceManager, PersistenceManagerInterface $persistenceManager)
```

Comment:
Why not inject resource manager and persistence manager via setter or property injection? You could move the constructor code to `getMappingResult()` to make sure that dependencies are already injected.

Reply:
I would need to keep a reference to the image, and I didn't want to do that. Also I prefer constructor injection very much for things that are not singleton.